card/lib/card/name.rb

```diff
@@ -34,19 +34,21 @@ def new str, validated_parts=nil
         str = str.to_s
         if !validated_parts && str.include?(joint)
           compose Cardname.split_parts(str)
-        elsif special_prefix?(str)
-          Card.fetch_name str # handles ~ and :
+        elsif (id = Card.id_from_string(str))  # handles ~ and :
+          Card.fetch_name(id) { Card.bad_mark(str) }
```

Comment:
it makes me a little uncomfortable that we raise such a loud error from bad data.  But I'm not sure I have a better idea...